src/store/participantsStore.js

```diff
@@ -338,30 +344,51 @@ const mutations = {
 			if (state.inCall[token] && state.inCall[token][sessionId]) {
 				Vue.delete(state.inCall[token], sessionId)
 			}
-
-			if (state.connecting[token] && state.connecting[token][sessionId]) {
-				Vue.delete(state.connecting[token], sessionId)
-			}
 		} else {
 			if (!state.inCall[token]) {
 				Vue.set(state.inCall, token, {})
 			}
 			Vue.set(state.inCall[token], sessionId, flags)
-
-			if (!state.connecting[token]) {
-				Vue.set(state.connecting, token, {})
-			}
-			Vue.set(state.connecting[token], sessionId, flags)
 		}
 	},
 
 	connectionFailed(state, { token, payload }) {
 		Vue.set(state.connectionFailed, token, payload)
 	},
 
+	clearConnectionFailed(state, token) {
+		Vue.delete(state.connectionFailed, token)
+	},
+
+	joiningCall(state, { token, sessionId, flags }) {
+		if (!state.joiningCall[token]) {
+			Vue.set(state.joiningCall, token, {})
+		}
+		Vue.set(state.joiningCall[token], sessionId, flags)
+	},
+
+	finishedJoiningCall(state, { token, sessionId }) {
+		if (state.joiningCall[token] && state.joiningCall[token][sessionId]) {
+			Vue.delete(state.joiningCall[token], sessionId)
+			if (!Object.keys(state.joiningCall[token]).length) {
+				Vue.delete(state.joiningCall, token)
+			}
+		}
+	},
+
+	connecting(state, { token, sessionId, flags }) {
+		if (!state.connecting[token]) {
+			Vue.set(state.connecting, token, {})
+		}
+		Vue.set(state.connecting[token], sessionId, flags)
+	},
+
 	finishedConnecting(state, { token, sessionId }) {
 		if (state.connecting[token] && state.connecting[token][sessionId]) {
 			Vue.delete(state.connecting[token], sessionId)
+			if (!Object.keys(state.connecting[token]).length) {
```

Comment:
Maybe we don't need to store them by token at all, since we can have only one session per tab?
But something for the follow-up, when migrate to pinia